rotkehlchen/exchanges/bybit.py

```diff
@@ -468,11 +484,79 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
                     msg = str(e)
                     if isinstance(e, KeyError):
                         msg = f'Missing key entry for {msg}.'
-                    return None, f'Error processing Bybit balance entry {coin_data}. {msg}'
+                    return {}, f'Error processing Bybit balance entry {coin_data}. {msg}'
 
                 assets_balance[asset] += Balance(amount=amount, usd_value=usd_value)
 
-        return assets_balance, ''
+        return assets_balance, None
+
+    def _query_funding_balances(self) -> tuple[dict[AssetWithOracles, Balance], str | None]:
+        """
+        Query balances in the funding wallet.
+        This call assumes that the first connection has been made to identify the account type.
+
+        Returns the a tuple of balances and None if there wasn't any error or a string
+        message with a description of what went wrong.
+        """
+        assets_balance: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        response, error = self._query_balances_or_error(
+            path='asset/transfer/query-account-coins-balance',
+            options={'accountType': 'FUND'},
+        )
+        if error is not None:
+            return {}, error
+
+        for entry in response.get('balance', []):
+            try:
+                asset = asset_from_bybit(entry['coin'])
```

Comment:
isn't this the same as account's balances inner loop? Can't you abstract it and call it in both functions?

Reply:
Not exactly the same. I've abstracted what could be done